### PR TITLE
PMM-4293 new collstats

### PR DESCRIPTION
**Pull request by percona-csalguero**
Replaced old `collStats` for the aggregation version `$collStats`

### CODE REVIEW

**Comment by BupycHuk** [comment]
No changes

File: exporter/collstats_collector.go [hunk 1]
```diff
@@ -20,23 +21,36 @@ func (d *collstatsCollector) Describe(ch chan<- *prometheus.Desc) {
 }
 
 func (d *collstatsCollector) Collect(ch chan<- prometheus.Metric) {
+	ctx := context.Background()
```
Comment (BupycHuk) [inline exporter/collstats_collector.go]: why don't use this? https://github.com/Percona-Lab/mnogo_exporter/pull/11/files#diff-a7b5017c3776f04bf91dede50897600fR14
Why do we need it at all?

File: exporter/collstats_collector.go [hunk 2]
```diff
@@ -20,23 +21,36 @@ func (d *collstatsCollector) Describe(ch chan<- *prometheus.Desc) {
...
+		if err != nil {
+			logrus.Errorf("cannot get $collstats for collection %s.%s: %s", database, collection, err)
 			continue
 		}
 
-		for _, metric := range buildMetrics(m) {
-			ch <- metric
+		var stats []bson.M
+		if err = cursor.All(ctx, &stats); err != nil {
+			panic(err)
```
Comment (BupycHuk) [inline exporter/collstats_collector.go]: Do you think that panic is a good idea here?

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: exporter/collstats_collector.go (hunk 2 above)
Comment (AlekSi) [reply exporter/collstats_collector.go]: No, definitely not. We should log error, but not crash.

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: exporter/collstats_collector.go (hunk 1 above)
Comment (AlekSi) [reply exporter/collstats_collector.go]: Created https://jira.percona.com/browse/PMM-6254 to track context/timeouts

File: exporter/exporter.go [hunk 3]
```diff
@@ -23,12 +23,17 @@ type Exporter struct {
...
-	Log  *logrus.Logger
+	CollStatsCollections []string
+	DSN                  string
+	Log                  *logrus.Logger
+	Path                 string
+	Port                 int
 }
 
+var (
+	errCannotHandleType = fmt.Errorf("don't know how to handle data type")
```
Comment (BupycHuk) [inline exporter/exporter.go]: you anyway wrap this error, so any reason to extract it as separate error?

**Review by BupycHuk — COMMENTED** [review]
please change base branch to make it easy to review.

File: exporter/exporter.go (hunk 3 above)
Comment (percona-csalguero) [reply exporter/exporter.go]: Linter and go errors for Go 1.13+

**Review by BupycHuk — APPROVED** [review]

File: exporter/collstats_collector.go (hunk 1 above)
Comment (AlekSi) [reply exporter/collstats_collector.go]: But that should be `context.TODO()`

File: exporter/diagnostic_data_collector.go [hunk 4]
```diff
@@ -38,7 +37,7 @@ func (d *diagnosticDataCollector) Collect(ch chan<- prometheus.Metric) {
 	var m bson.M
 
 	cmd := bson.D{{Key: "getDiagnosticData", Value: "1"}}
-	res := d.client.Database("admin").RunCommand(d.ctx, cmd)
+	res := d.client.Database("admin").RunCommand(context.Background(), cmd)
```
Comment (AlekSi) [inline exporter/diagnostic_data_collector.go]: `context.TODO()`

**Review by AlekSi — APPROVED** [review]

**Pull request merged**